pkg/metrics/exporter.go

```diff
@@ -48,20 +52,30 @@ func RegisterPrometheusExporter(registry *prom.Registry) (http.Handler, error) {
 
 // RegisterStackdriverExporter register a Stackdriver exporter to OpenCensus.
 // It will add Agones metrics into Stackdriver on Google Cloud.
-func RegisterStackdriverExporter(projectID string) (sd *stackdriver.Exporter, err error) {
-	// Default project will be used
-	sd, err = stackdriver.NewExporter(stackdriver.Options{
+func RegisterStackdriverExporter(projectID string, defaultLabels string) (*stackdriver.Exporter, error) {
+	monitoredRes, err := getMonitoredResource(projectID)
+	if err != nil {
+		logger.WithError(err).Warn("error discovering monitored resource")
```

Comment:
Is it ok if monitoredRes is nil later on or does this need to be followed by a `return nil, err` line?

Reply:
no it's ok this is a best effort.

Reply:
by default it is nil.